src/cmap/manager.rs

```diff
@@ -0,0 +1,68 @@
+use tokio::sync::mpsc;
+
+use super::Connection;
+use crate::error::Error;
+
+#[derive(Debug)]
+pub(super) enum PoolManagementRequest {
+    Clear,
+    Populate(Connection),
+    CheckIn(Connection),
+    HandleConnectionFailed(Error),
+}
+
+impl PoolManagementRequest {
+    fn unwrap_check_in(self) -> Connection {
```

Comment:
It's not clear to me where the `unwrap_` methdods are used (for this type or the other request types). Are they just used in tests, and if so, can we annotate them as such?

Reply:
Can you walk me through why we know these won't panic in the cases that they're called? There are a lot of moving pieces, so I'm not sure I fully understand

Reply:
Never mind, I think I understand now